src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs

```diff
@@ -1467,20 +1467,26 @@ private void DynamoViewModelRequestAboutWindow(DynamoViewModel model)
         private void DynamoViewModelRequestRequestPackageManagerPublish(PublishPackageViewModel model)
         {
             var cmd = Analytics.TrackCommandEvent("PublishPackage");
-            if (_pubPkgView == null)
+
+            if (packageManagerWindow == null)
             {
-                _pubPkgView = new PublishPackageView(model)
+                packageManagerWindow = new PackageManagerView(this, _pkgVM)
                 {
                     Owner = this,
                     WindowStartupLocation = WindowStartupLocation.CenterOwner
                 };
-                _pubPkgView.Closed += (sender, args) => { _pubPkgView = null; cmd.Dispose(); };
-                _pubPkgView.Show();
 
-                if (_pubPkgView.IsLoaded && IsLoaded) _pubPkgView.Owner = this;
+                // setting the owner to the packageManagerWindow will centralize promts originating from the Package Manager
+                dynamoViewModel.Owner = packageManagerWindow;
```

Comment:
This line does not look correct to me... did I miss anything?

Reply:
The way I used it and the way I understand it is, whichever more major **View** 'takes control', gets ownership of this property. The same thing happens in the PreferencesView: 

https://github.com/DynamoDS/Dynamo/blob/caa3268d4c813a13c830faf402d1a8a89073954a/src/DynamoCoreWpf/Views/Menu/PreferencesView.xaml.cs#L77

The `Owner` property is used when utilizing the MessageBoxService:

`public static MessageBoxResult Show(Window owner,string msg, string title, MessageBoxButton button, MessageBoxImage img)` 

So when we launch message boxes from inside the DynamoViewModel, we are centralizing them around that `Owner`. 

Another thing it does is to keep the reference of the current window high up in the hierarchy. For example, in one instance, we are calling the `UnninstallCommand` which is part of the PackageViewModel from deep inside the PacakgeManagerView. The PackageViewModel and the `Uninstall` method do not have a reference to the PackageManagerView, though, but they do have a reference to the DynamoViewModel. So when prompting the user, we are setting the `Owner` of the DynamoViewModel, which is the PackageManagerView, so the prompt is correctly centralized. 

I did not do a good job of cleaning up this logic, the logic exists from before. Maybe I did not fully understand it myself, this is my current working logic of it.

Reply:
After checking out the code, I think I understand what you are saying. I believe the Owner property was created so DynamoViewModel could have a reference to the view as a convenient way to raise message box with the Owner correctly passed in. But it is not meant to be served as the property indicating current Window, because I do not see any logic to reset that Owner property back to DynamoView. I can foresee a bug that after Preferences panel opened and dragged out of the screen, then closed, new message boxes using such owner will be out of DynamoView. 

But thank you for pointing this out, I do think we provided bad examples in code so it's good for this PR but I will file a follow up task to address that.

Reply:
I understand. If you want me, I can try to trace all the moments where we are setting using message boxes and try to remove that need by setting the Onwer directly somehow.